crypto/store/loader_file.c

```diff
@@ -256,53 +255,57 @@ static OSSL_STORE_INFO *try_decode_PKCS12(const char *pem_name,
                 OSSL_STORE_INFO *osi_pkey = NULL;
                 OSSL_STORE_INFO *osi_cert = NULL;
                 OSSL_STORE_INFO *osi_ca = NULL;
-
-                if ((ctx = sk_OSSL_STORE_INFO_new_null()) != NULL
-                    && (osi_pkey = OSSL_STORE_INFO_new_PKEY(pkey)) != NULL
-                    && sk_OSSL_STORE_INFO_push(ctx, osi_pkey) != 0
-                    && (osi_cert = OSSL_STORE_INFO_new_CERT(cert)) != NULL
-                    && sk_OSSL_STORE_INFO_push(ctx, osi_cert) != 0) {
-                    ok = 1;
-                    osi_pkey = NULL;
-                    osi_cert = NULL;
-
-                    while(sk_X509_num(chain) > 0) {
+                int ok = 1;
+
+                if ((ctx = sk_OSSL_STORE_INFO_new_null()) != NULL) {
+                    if (pkey != NULL) {
+                        if ((osi_pkey = OSSL_STORE_INFO_new_PKEY(pkey)) != NULL
+                            && (pkey = NULL) == NULL
```

Comment:
This construct is 'interesting'. On the other hand the intention is clear, so I just hope that Coverity won't barf on it.

Reply:
> This construct is 'interesting'. On the other hand the intention is clear, so I just hope that Coverity won't barf on it.

Yeah, I'm proud about this nice trick avoiding lengithy case distinctions :smile:

Reply:
It's tricky code, and isn't something to be proud about.  It should be rewritten.  As that is unlikely to happen :(, please comment it.

Reply:
All right, added an explaining comment in both instances.